src/routes/(app)/api/search/sync/+server.ts

```diff
@@ -0,0 +1,118 @@
+import type { RequestHandler } from "./$types";
+
+export const GET: RequestHandler = async ({ locals, getClientAddress }) => {
+  // Deny access if not localhost
+  const ip = getClientAddress();
+  if (!(ip === "localhost" || ip === "::1" || ip === "127.0.0.1")) {
+    return new Response("Unauthorized", { status: 401 });
+  }
+  const prisma = locals.prisma;
+
+  const members = await prisma.member.findMany({
+    select: {
+      id: true,
+      studentId: true,
+      firstName: true,
+      lastName: true,
+      nickname: true,
+    },
+  });
+  const songs = await prisma.song.findMany({
+    select: {
+      id: true,
+      title: true,
+      category: true,
+      lyrics: true,
+      melody: true,
+    },
+    where: {
+      deletedAt: null,
+    },
+  });
+  const articles = await prisma.article.findMany({
+    select: {
+      id: true,
+      body: true,
+      bodyEn: true,
+      header: true,
+      headerEn: true,
+    },
+    where: {
+      AND: [
+        {
+          removedAt: null,
+        },
+        {
+          publishedAt: {
+            not: null,
+          },
+        },
+      ],
+    },
+  });
+  const events = await prisma.event.findMany({
+    select: {
+      id: true,
+      title: true,
+      titleEn: true,
+      description: true,
+      descriptionEn: true,
+    },
+    where: {
+      AND: [
+        {
+          removedAt: null,
+        },
```

Comment:
This is not needed - our access policies already ensure this.

Reply:
Why not double protection? :)